assignment/tool/src/webapp/js/assignments.js

```diff
@@ -1006,6 +1006,92 @@ ASN.disableTimesheetSetupSection = function()
     ASN.toggleAutoAnnounceEstimate(false);
 }
 
+ASN.autocompleteRubricWithSelfReport = function(event) {
+    event.preventDefault();
+
+    const studentRubric = document.querySelector("sakai-rubric-student");
+    if (!studentRubric) {
+        console.error("sakai-rubric-student element not found");
+        return;
+    }
+
+    const gradingRubric = document.querySelector("sakai-rubric-grading");
+    if (!gradingRubric) {
+        console.error("sakai-rubric-grading element not found");
+        return;
+    }
+
+    const evaluationDetails = [];
+    const criterionRows = studentRubric.querySelectorAll(".criterion-row");
+    criterionRows.forEach(row => {
+        const criterionId = row.id.split("_").pop();
+        const selectedRating = row.querySelector(".rating-item.selected");
+        if (selectedRating) {
+            const selectedRatingId = selectedRating.id.split("-").pop();
+            evaluationDetails.push({ criterionId, selectedRatingId });
+        }
+    });
```

Comment:
```suggestion
    const evaluationDetails = [...studentRubric.querySelectorAll(".criterion-row")].reduce(acc, row) => {
    
        const selectedRatingId = row.querySelector(".rating-item.selected")?.id.split("-").pop();
        selectedRatingId && acc.push({ criterionId: row.id.split("_").pop(), selectedRatingId });
        return acc;
    }, []);
```

Reply:
I'm spreading the NodeList into an array and then reducing the array. I'm using optional chaining to remove some of the variables you had introduced.